xdsl/ir/core.py

```diff
@@ -426,22 +426,13 @@ def print_parameter(self, printer: Printer):
 class ParametrizedAttribute(Attribute):
     """An attribute parametrized by other attributes."""
 
-    def __init__(self, *parameters: Attribute):
-        if len(parameters) == 1 and isinstance(parameters[0], tuple):
-            import warnings
-
-            warnings.warn(
-                "Passing a tuple as a single argument to ParametrizedAttribute.__init__ is deprecated. "
-                "Pass the tuple elements as separate arguments instead.",
-                DeprecationWarning,
-                stacklevel=2,
-            )
-            parameters = parameters[0]
-
-        for (f, _), param in zip(
-            self.get_irdl_definition().parameters, parameters, strict=True
-        ):
-            object.__setattr__(self, f, param)
+    def __init__(self, *parameters: Any):
+        irdl_def = self.get_irdl_definition()
```

Comment:
Removed the deprecated thing as it was throwing pyright errors, but could probably figure it out to keep it around a bit longer

Reply:
Adding context from zulip. Since its now valid for a parameter to be a tuple (which gets converted) there's no way to keep this warning without breaking stuff. I believe the plan is to wait a release, then remove the warning, then add converters.